src/toast/examples.js

```diff
@@ -0,0 +1,207 @@
+  position: 'relative',
+  display: 'flex',
+  flexDirection: 'column',
+  justifyContent: 'center',
+  alignItems: 'center',
+  lineHeight: 1.5,
+});
+const Space = styled('span', {
+  display: 'inline-block',
+  lineHeight: 1.5,
+  marginTop: '10px',
+  marginRight: '10px',
+  marginLeft: '10px',
+});
+
+const getBtnStyle = color => ({$theme}) => {
+  const hoverStyle = {backgroundColor: $theme.colors[`${color}500`]};
+  const activeStyle = {backgroundColor: $theme.colors[`${color}600`]};
+  return {
+    backgroundColor: $theme.colors[color],
+    ':hover:enabled': hoverStyle,
+    ':focus:enabled': hoverStyle,
+    ':active:enabled': activeStyle,
+  };
+};
+
+class NotificationExample extends React.Component<{}, {toasts: []}> {
+  state = {
+    toasts: [],
+  };
+  toastId = 0;
+
+  getToastProps(props) {
+    const toastId = this.toastId++;
+    return {
+      key: toastId,
+      autoHideDuration: 4000,
+      onClose: this.getOnCloseHandler(toastId),
+      children: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
+      ...props,
+    };
+  }
+
+  add = props => () => {
+    const toastProps = this.getToastProps(props);
+    this.setState(state => {
+      const toasts = [...state.toasts];
+      toasts.unshift(toastProps);
+      return {toasts};
+    });
+  };
+
+  addInfo = this.add({});
+  addPositive = this.add({kind: KIND.positive});
+  addWarning = this.add({kind: KIND.warning});
+  addNegative = this.add({kind: KIND.negative});
+
+  dismiss = (key: number) => {
+    this.setState(({toasts}) => ({
+      // $FlowFixMe
+      toasts: toasts.filter(t => {
+        return !(t.key === key);
+      }),
+    }));
+  };
+
+  getOnCloseHandler(key) {
+    return () => {
+      this.dismiss(key);
+    };
+  }
+
+  render() {
+    return (
+      <div>
+        <Button onClick={this.addInfo}>Info toast</Button>
+        <Space />
+        <Button
+          onClick={this.addPositive}
+          overrides={{
+            BaseButton: {
+              style: getBtnStyle('positive'),
+            },
+          }}
+        >
+          Success toast
+        </Button>
+        <Space />
+        <Button
+          onClick={this.addWarning}
+          overrides={{
+            BaseButton: {
+              style: getBtnStyle('warning'),
+            },
+          }}
+        >
+          Warning toast
+        </Button>
+        <Space />
+        <Button
+          onClick={this.addNegative}
+          overrides={{
+            BaseButton: {
+              style: getBtnStyle('negative'),
+            },
+          }}
+        >
+          Negative toast
+        </Button>
+        <Centered>
+          {this.state.toasts.map((toastOptions, index) => {
+            return <Toast key={index} {...toastOptions} />;
+          })}
+        </Centered>
+      </div>
+    );
+  }
+}
+
+export default {
+  [tests.SIMPLE_EXAMPLE]: function Story1() {
+    return (
+      <Centered>
+        <Toast>Default info notification</Toast>
```

Comment:
every story contain many examples. Afaik, we still use stories for e2e tests as basic. If we start writing e2e tests , it may be not easy to test story as black-boxed Selenium will not be able to pick a single example and work only with it without providing extra props and other identifiers

Reply:
I think we need to optimize examples for better stories and documentation in the first place and figure out tests as a second. For instance, maybe create additional examples for e2e tests only.

Reply:
yes, this is reasonable. However, the main idea of shared examples file is to re-use it for both stories and e2e tests. Having different examples may duplicate our code. There is not single approach on how to optimize examples for customer better. Some provide examples for every single property. Others gather some examples under "one roof"
https://mineral-ui.com/components/text-input
https://material-ui.com/demos/text-fields/
https://ant.design/components/input/
https://react.semantic-ui.com/elements/input/
https://grommet.io/docs/search